src/screens/planActivity/PlanFormControls.tsx

```diff
@@ -1,41 +1,45 @@
 import { FormikProps } from 'formik';
 import React, { SFC } from 'react';
 import { StyleSheet, View } from 'react-native';
+import { NavigationInjectedProps, withNavigation } from 'react-navigation';
 
 import PlayButton, { TextInput } from 'components';
 import { i18n } from 'locale';
-import { Plan } from 'models';
-import { dimensions, getElevation, palette } from 'styles';
+import { dimensions, palette } from 'styles';
 import { Icon, StyledText } from '../../components';
 import { PlanFormData } from './PlanForm';
 import { ShuffleButton } from './ShuffleButton';
 
-interface Props extends FormikProps<PlanFormData> {
-  plan: Plan;
-}
+interface Props extends FormikProps<PlanFormData>, NavigationInjectedProps {}
 
-export const PlanFormControls: SFC<Props> = ({ values, handleChange, isSubmitting, submitForm, plan }) => (
-  <View style={styles.container}>
-    <View style={styles.inputContainer}>
-      <Icon name="emoticon" size={24} color={palette.textInputPlaceholder} />
-      {isSubmitting ? (
-        <StyledText style={styles.styledText}>{values.planInput}</StyledText>
-      ) : (
-        <TextInput
-          style={styles.textInput}
-          placeholder={i18n.t('planActivity:planNamePlaceholder')}
-          value={values.planInput}
-          onChangeText={handleChange('planInput')}
-          onEndEditing={submitForm}
-        />
-      )}
-    </View>
-    <View style={styles.buttonContainer}>
-      <ShuffleButton disabled={!isSubmitting} />
-      <PlayButton plan={plan} disabled={!isSubmitting} size={50} />
+export const PlanFormControls: SFC<Props> = ({ values, handleChange, isSubmitting, submitForm, ...props }) => {
```

Comment:
We need to change the name for the flag `isSubmitting` since it is even valid for the scenario when form is submitted. Naming is confusing here.